fmriprep/workflows/bold/base.py

```diff
@@ -279,11 +278,23 @@ def init_func_preproc_wf(bold_file, ignore, freesurfer,
                 't1_2_fsnative_forward_transform', 't1_2_fsnative_reverse_transform']),
         name='inputnode')
 
-    if isinstance(bold_file, list):
-        inputnode.inputs.iterables = ("bold_file", bold_file)
+    if multiecho:  # if multi-echo data
+        # iterate most preprocessing steps over each echo
+        inputnode.iterables = ('bold_file', bold_file)
+
+        me_first_echo = pe.JoinNode(interface=FirstEcho(),
```

Comment:
I think this would be more readable if the interface instantiation happens right before it is connected (~line 460)